pkg/trace/stats/span_concentrator.go

```diff
@@ -184,24 +184,28 @@ func (sc *SpanConcentrator) addSpan(s *StatSpan, aggKey PayloadAggregationKey, c
 	b, ok := sc.buckets[btime]
 	if !ok {
 		b = NewRawBucket(uint64(btime), uint64(sc.bsize))
-		if containerID != "" && len(containerTags) > 0 {
-			b.containerTagsByID[containerID] = containerTags
-		}
 		sc.buckets[btime] = b
 	}
+	if tags.processTagsHash != 0 && len(tags.processTags) > 0 {
+		b.processTagsByHash[tags.processTagsHash] = tags.processTags
+	}
+	if tags.containerID != "" && len(tags.containerTags) > 0 {
+		b.containerTagsByID[tags.containerID] = tags.containerTags
+	}
 	b.HandleSpan(s, weight, origin, aggKey)
 }
 
 // AddSpan to the SpanConcentrator, appending the new data to the appropriate internal bucket.
-func (sc *SpanConcentrator) AddSpan(s *StatSpan, aggKey PayloadAggregationKey, containerID string, containerTags []string, origin string) {
-	sc.addSpan(s, aggKey, containerID, containerTags, origin, 1)
+func (sc *SpanConcentrator) AddSpan(s *StatSpan, aggKey PayloadAggregationKey, tags infraTags, origin string) {
```

Comment:
Note that this is also accessed from `dd-trace-go`, thus its code needs to be updated after upgrading the agent version.

Reply:
If this is accessed by `dd-trace-go` does the `infraTags` struct needs to be public?